Added discoIPC rich presence wrapper (Python)

k3rn31p4nic opened the pull request.

msciotti commented:
Hi @k3rn31p4nic! Thank you for your contribution of this wrapper. Before I can merge, I would ask for a couple things.

First, please add your wrapper to the list in alphabetic order by language name. So, below the Kotlin wrapper.

Secondly, this implementation does not fully wrap the functionality of the SDK. Much like the other python implementation currently PR'd, you'll need to add a way to handle the events for joining, spectating, and ask to join.

If you're interested in adding these, I'm happy to keep the PR open and merge when added!

k3rn31p4nic commented:
I thought the alphabetical order was based on the name of the library, that's why I added it there.

I didn't originally add support for joining, spectating and ask to join events because I didn't needed that in my project for which I actually made this wrapper. But I tried adding support for it today, but it didn't worked for me and my testers. The rich presence is updated, it even shows the **Ask to Join** & **Spectate** button, but none of us are able to click on it in the Discord client. And in the Join/Spectate invite embed, it shows that the game is not detected, but the user is playing it and it's also showing up in the user's profile. I tried a lot, but I'm not able to find the cause.
While I want to support this in my wrapper, I'm not able to do it because of that (or maybe because I'm an idiot) and I also don't have enough time to work on this currently as I've other projects to work on. But I will add these in the future, but that's not anytime soon.

So, if that's a requirement for this, this PR should be closed and I'll open a new PR when I've added support for it.

Thanks.

msciotti commented:
Thanks for being understanding. In order to detect properly, you'll need to handle application protocol registration as well. That's what lets Discord know what games you can and cannot launch. You can check it out at https://github.com/discordapp/discord-rpc/blob/master/src/discord_register_win.cpp and the corresponding macOS and linux files.

I welcome a re-opened PR if you find that you want the time!
